### PR TITLE
[Feature][GCS FT] Best-effort redis cleanup job

### PR DESCRIPTION
**Pull request by rueian**
## Why are these changes needed?

Following the previous proposal https://github.com/ray-project/kuberay/issues/1557#issuecomment-1784162985, this PR introduces a new environment variable `GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS` to the kuberay-operator.

This variable will be set to the `.spec.activeDeadlineSeconds` option of the redis cleanup job of a fault-tolerant RayCluster, and it will terminate the job after the deadline no matter what.

This PR further makes the kuberay-operator remove the `ray.io/gcs-ft-redis-cleanup-finalizer` finalizer on the RayCluster allowing it to be deleted after the job is terminated.

The default value of `GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS` is 300. This essentially makes all redis cleanup jobs work in a best-effort manner for 5 minutes by default.

## Related issue number

https://github.com/ray-project/kuberay/issues/1725
https://github.com/ray-project/kuberay/issues/1557

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/raycluster_controller.go
Comment (kevin85421) [inline ray-operator/controllers/ray/raycluster_controller.go:288]: (You can ignore this comment if we decide to merge the logic.)
Here, we may only need to check for `batchv1.JobFailed`. The K8s Job is a non-parallel Job, so the Job is complete as soon as its Pod terminates successfully ([ref](https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs)).

File: ray-operator/controllers/ray/utils/util.go
Comment (kevin85421) [inline ray-operator/controllers/ray/utils/util.go:513]: (You can ignore this comment if we decide to merge the logic.)
If we only need to check `batchv1.JobFailed`, we may be able to remove this function.

File: ray-operator/controllers/ray/raycluster_controller.go
Comment (kevin85421) [inline ray-operator/controllers/ray/raycluster_controller.go:287]: Can we merge this `if` statement and the `if redisCleanupJob.Status.Succeeded > 0` statement?

```
if utils.IsJobFinished(&redisCleanupJob) {
  isBestEffort := redisCleanupJob.Spec.ActiveDeadlineSeconds != nil
  if (Job is complete) {
    print some logs
  } else {
    print some logs
    maybe file a k8s event
  }
  remove finalizer
}
```

File: ray-operator/controllers/ray/raycluster_controller.go
Comment (kevin85421) [inline ray-operator/controllers/ray/raycluster_controller.go:185]: This variable seems only to be used by `buildRedisCleanupJob`. We can read the env var in `buildRedisCleanupJob`.

File: ray-operator/controllers/ray/raycluster_controller.go
Comment (kevin85421) [inline ray-operator/controllers/ray/raycluster_controller.go:1123]: Maybe we don't need this variable because we can read the env var here.

File: ray-operator/controllers/ray/utils/util.go
Comment (kevin85421) [inline ray-operator/controllers/ray/utils/util.go:525]: Perhaps we could directly use `strconv.ParseInt(v, 10, 64)` in the `buildRedisCleanupJob` function. If users input invalid values, it should fail. In open-source projects, we often need to address numerous user questions without access to their Kubernetes clusters. Therefore, identifying and surfacing issues as early as possible might be a better strategy for maintainers.

Comment (rueian) [reply ray-operator/controllers/ray/utils/util.go:525]: Hi @kevin85421,

I find it hard to make the operator fail in the `buildRedisCleanupJob`. This function is called only when the operator is going to create the cleanup job, which is already too late.

Considering that the `GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS` is an environment variable, the proper timing to fail the operator should be during initialization. How about we parse the env and set the parsed result as a member field of RayClusterReconciler?

File: ray-operator/controllers/ray/raycluster_controller.go
Comment (rueian) [reply ray-operator/controllers/ray/raycluster_controller.go]: Done.

File: ray-operator/controllers/ray/utils/util.go
Comment (rueian) [reply ray-operator/controllers/ray/utils/util.go]: After our latest discussion, we decided not to introduce the `GCS_FT_REDIS_CLEANUP_DEADLINE_SECONDS` but always set the `.spec.activeDeadlineSeconds` to 300 seconds instead.

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**
